File: 2_fibonacci/fibonacci_dynamic.c
#include <stdio.h>
#include <stdlib.h>


void fibonacci(int n) {
  int arr[2], k, temp;
  arr[0] = 0;
  arr[1] = 1;
  for (int i = 0 ; i < n ; i++) {
    k = arr[0] + arr[1];
    temp = arr[1];
    arr[0] = temp;
    arr[1] = k;
    printf('%d', k);
  }
  return;
}

int main() {
  fibonacci(10);
  return 0;  
}

File: 2_fibonacci/fibonacci_dynamic.py
# coding utf8
# Python implemtation of the Fibonacci Series with dynamic programming
# Author : Quentin Burg


def fibonacci(n):
    arr = [0,1]
    for i in range (n):
        k = arr[1] + arr[0]
        print(k, end=" ")
        arr.append(k)
        del arr[0]
    return k

fibonacci(10)
        
